src/xtgeo/well/_wellmarkers.py

```diff
@@ -70,12 +74,20 @@ def get_zonation_points(self, tops, incl_limit, top_prefix, zonelist, use_undef)
 
     zlist = ztops if tops else zisos
 
-    logger.debug(zlist)
+    # collect temporary columns, made by geometrics() and _extract_ztops() methods
+    tmp_columns = [
+        col
+        for col in (ztopnames or []) + (zisonames or [])
+        if col.startswith("Q_") and col not in self.get_lognames()
+    ]
 
-    if tops:
-        return pd.DataFrame(zlist, columns=ztopnames)
+    dataframe = (
+        pd.DataFrame(zlist, columns=ztopnames)
+        if tops
+        else pd.DataFrame(zlist, columns=zisonames)
+    )
 
-    return pd.DataFrame(zlist, columns=zisonames)
+    return dataframe.drop(columns=tmp_columns, errors="ignore")
```

Comment:
This appears as "beyond bug" change, but it was never the intention to add temporary "working" columns to the dataframes (columns describing geometry, starting with "Q_", like "Q_AZI"). Hence they are removed here.